cpp/include/rmm/cuda_stream.hpp

```diff
@@ -38,6 +38,14 @@ namespace RMM_EXPORT rmm {
  */
 class cuda_stream {
  public:
+  /**
+   * @brief stream creation flags.
+   */
+  enum class flags : unsigned int {
+    sync_default = cudaStreamDefault,  ///< Created stream synchronises with the default stream.
+    non_blocking =
+      cudaStreamNonBlocking,  ///< Created stream does not synchronise with the default stream.
```

Comment:
We use the US form of the word in RMM.
```
$ rg "synchroniz" | wc -l
107
$ rg "synchronis" | wc -l
0
```

```suggestion
    sync_default = cudaStreamDefault,  ///< Created stream synchronizes with the default stream.
    non_blocking =
      cudaStreamNonBlocking,  ///< Created stream does not synchronize with the default stream.
```

Reply:
Addrezzed review.